discovery/syncer.go

```diff
@@ -767,15 +764,21 @@ func (g *GossipSyncer) synchronizeChanIDs() (bool, error) {
 	log.Infof("GossipSyncer(%x): querying for %v new channels",
 		g.cfg.peerPub[:], len(queryChunk))
 
+	// Change the state before sending the query msg.
+	g.setSyncState(waitingQueryChanReply)
```

Comment:
but arn't we then in the wrong state for the case where `sendToPeer` below errors?

Reply:
ah, nvm, I see that is pre-existing behaviour